src/Cmfd.cpp

```diff
@@ -606,311 +539,140 @@ FP_PRECISION Cmfd::computeKeff(int moc_iteration) {
-        value = (material->getDifHat()[1*_num_cmfd_groups + e]
-                + material->getDifTilde()[1*_num_cmfd_groups + e])
-                * _cell_width;
-        
-        _A[cell][e*(_num_cmfd_groups+4)+e+2] += value;
+        value = (material->getDifHat()[SURFACE_Y_MIN*_num_cmfd_groups + e]
+                 + material->getDifTilde()[SURFACE_Y_MIN*_num_cmfd_groups + e])
+          * _cell_width;
+        _A->incrementValue(cell, e, cell, e, value);
 
         /* Set transport term on off diagonal */
         if (y != 0) {
-          value = - (material->getDifHat()[1*_num_cmfd_groups + e]
-                  - material->getDifTilde()[1*_num_cmfd_groups + e])
-              * _cell_width;
-          
-          _A[cell][e*(_num_cmfd_groups+4)+1] += value;
+          value = - (material->getDifHat()[SURFACE_Y_MIN*_num_cmfd_groups + e]
+                     - material->getDifTilde()
+                     [SURFACE_Y_MIN*_num_cmfd_groups + e]) * _cell_width;
+          _A->incrementValue(cell-_num_x, e, cell, e, value);
         }
 
-        /* TOP SURFACE */
+        /* SURFACE_Y_MAX */
 
         /* Set transport term on diagonal */
-        value = (material->getDifHat()[3*_num_cmfd_groups + e]
-                - material->getDifTilde()[3*_num_cmfd_groups + e])
-            * _cell_width;
-        
-        _A[cell][e*(_num_cmfd_groups+4)+e+2] += value;
+        value = (material->getDifHat()[SURFACE_Y_MAX*_num_cmfd_groups + e]
+                 - material->getDifTilde()[SURFACE_Y_MAX*_num_cmfd_groups + e])
+          * _cell_width;
+        _A->incrementValue(cell, e, cell, e, value);
 
         /* Set transport term on off diagonal */
         if (y != _num_y - 1) {
-          value = - (material->getDifHat()[3*_num_cmfd_groups + e]
-                  + material->getDifTilde()[3*_num_cmfd_groups + e])
-                  * _cell_width;
-          
-          _A[cell][e*(_num_cmfd_groups+4)+_num_cmfd_groups+3] += value;
+          value = - (material->getDifHat()[SURFACE_Y_MAX*_num_cmfd_groups + e]
+                     + material->getDifTilde()
+                     [SURFACE_Y_MAX*_num_cmfd_groups + e]) * _cell_width;
+          _A->incrementValue(cell+_num_x, e, cell, e, value);
         }
 
         /* Source term */
         for (int g = 0; g < _num_cmfd_groups; g++) {
           value = material->getChi()[e] * material->getNuSigmaF()[g]
                   * volume;
-
-          _M[cell][e*_num_cmfd_groups+g] += value;
+          _M->incrementValue(cell, g, cell, e, value);
         }
-
-        log_printf(DEBUG, "cell: %d, vol; %f", cell, volume);
-
-        for (int i = 0; i < _num_cmfd_groups+4; i++)
-          log_printf(DEBUG, "i: %d, A value: %f",
-                     i, _A[cell][e*(_num_cmfd_groups+4)+i]);
-
-        for (int i = 0; i < _num_cmfd_groups; i++)
-          log_printf(DEBUG, "i: %d, M value: %f",
-                     i, _M[cell][e*(_num_cmfd_groups)+i]);
       }
     }
```

Comment:
All `log_printf` statements in the current branch of develop indent line continuations to the start to this location so I'd recommend we change all `log_printf` statements at the same time (in a future PR) to remain consistent.